### PR TITLE
Deprecate `Trial.system_attrs` property method

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

Follow-up #4188 

## Description of the changes

Make `Trial.system_attrs` property deprecated.

### CODE REVIEW

File: optuna/trial/_trial.py [hunk 1]
```diff
@@ -699,13 +699,16 @@ def user_attrs(self) -> Dict[str, Any]:
...
+    @deprecated_func("3.1.0", "6.0.0")
     def system_attrs(self) -> Dict[str, Any]:
         """Return system attributes.
 
         Returns:
             A dictionary containing all system attributes.
         """
 
+        # TODO(c-bata): Remove the following assertion before merged the PR.
+        assert False, "Must not reach here on unit tests"
```
Comment (c-bata) [inline optuna/trial/_trial.py:711]: Note, this is just a temporary change to make it easier to detect all places that calls this property method.

```suggestion
```

**Review by contramundum53 — APPROVED** [review]
LGTM after removing the temporary change.

**Comment by c-bata** [comment]
Thanks for the quick review. I removed a temporary change.

**Comment by contramundum53** [comment]
@gen740 Could you review this PR?

**Review by gen740 — APPROVED** [review]
LGTM!

**Pull request merged**